homeassistant/components/keba/sensor.py

```diff
@@ -0,0 +1,122 @@
+"""Support for KEBA charging station sensors."""
+import logging
+
+from homeassistant.const import ENERGY_KILO_WATT_HOUR
+from homeassistant.helpers.entity import Entity
+from homeassistant.const import DEVICE_CLASS_POWER
+
+from . import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_platform(hass, config,
+                               async_add_entities, discovery_info=None):
+    """Set up the KEBA charging station platform."""
+    if discovery_info is None:
+        return
+
+    keba = hass.data[DOMAIN]
+
+    sensors = [
+        KebaSensor(keba, 'Curr user', 'Max current', 'mdi:flash',
+                   'A'),
+        KebaSensor(keba, 'Setenergy', 'Energy target', 'mdi:gauge',
+                   ENERGY_KILO_WATT_HOUR),
+        KebaSensor(keba, 'P', 'Charging power', 'mdi:flash',
+                   'kW', DEVICE_CLASS_POWER),
+        KebaSensor(keba, 'E pres', 'Session energy', 'mdi:gauge',
+                   ENERGY_KILO_WATT_HOUR),
+        KebaSensor(keba, 'E total', 'Total Energy', 'mdi:gauge',
+                   ENERGY_KILO_WATT_HOUR)
+    ]
+    async_add_entities(sensors)
+
+
+class KebaSensor(Entity):
+    """The entity class for KEBA charging stations sensors."""
+
+    def __init__(self, keba, key, name, icon, unit, device_class=None):
+        """Initialize the KEBA Sensor."""
+        self._key = key
+        self._keba = keba
+        self._name = name
+        self._device_class = device_class
+        self._icon = icon
+        self._unit = unit
+        self._state = None
+        self._attributes = {}
+
+    @property
+    def should_poll(self):
+        """Deactivate polling. Data updated by KebaHandler."""
+        return False
+
+    @property
+    def unique_id(self):
+        """Return the unique ID of the binary sensor."""
+        return f"{self._keba.device_name}_{self._name}"
+
+    @property
+    def name(self):
+        """Return the name of the device."""
+        return self._name
+
+    @property
+    def device_class(self):
+        """Return the class of this sensor."""
+        return self._device_class
+
+    @property
+    def icon(self):
+        """Icon to use in the frontend, if any."""
+        return self._icon
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        return self._state
+
+    @property
+    def unit_of_measurement(self):
+        """Get the unit of measurement."""
+        return self._unit
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes of the binary sensor."""
+        return self._attributes
+
+    async def async_update(self):
+        """Get latest cached states from the device."""
+        self._state = self._keba.get_value(self._key)
+
+        if self._key == 'P':
+            self._attributes['power_factor'] = \
+                self._keba.get_value('PF')
+            self._attributes['voltage_U1'] = \
```

Comment:
Please use lowercase for the attribute names.

Reply:
oh, I missed these. Will be fixed in next commit